### PR TITLE
Update jdbc driver version to 42.7.3-yb-1

### PR DESCRIPTION
**Pull request by ashetkar**
- Update jdbc driver version to 42.7.3-yb-1
- Few updates to the content on node-type aware load balancing

### CODE REVIEW

File: docs/content/v2.20/drivers-orms/java/yugabyte-jdbc.md [hunk 1]
```diff
@@ -94,12 +94,55 @@ Install the added dependency using `mvn install`.
...
 
 ```java
-implementation 'com.yugabyte:jdbc-yugabytedb:42.3.5-yb-8'
+implementation 'com.yugabyte:jdbc-yugabytedb:42.7.3-yb-1'
 implementation 'com.zaxxer:HikariCP:4.0.3'
 ```
 
 ### Step 2: Set up the database connection
 
+<table>
```
Comment (ddhodge) [inline docs/content/v2.20/drivers-orms/java/yugabyte-jdbc.md:103]: Can we not just xref to the driver page for this

File: docs/content/stable/drivers-orms/java/yugabyte-jdbc.md [hunk 2]
```diff
@@ -117,18 +117,18 @@ The following table describes the connection parameters required to connect, inc
...
+| `user` | User connecting to the database | yugabyte |
+| `password` | User password | yugabyte |
+| `load-balance` | Enables [Uniform load balancing](../../smart-drivers/#cluster-aware-load-balancing) | false (Disabled) |
+| `yb-servers-refresh-interval` | The interval in seconds to refresh the servers list; ignored if `load-balance` is false | 300 |
+| `topology-keys` | Enables [Topology-aware load balancing](../../smart-drivers/#topology-aware-load-balancing). It can be set to comma-separated geo-locations in the form `cloud.region.zone:priority`. Ignored if `load-balance` is false | Empty |
+| `fallback-to-topology-keys-only` | If set to true and `topology-keys` are specified, the driver only tries to connect to nodes specified in `topology-keys` | false |
+| `failed-host-reconnect-delay-secs` | When the driver is unable to connect to a node, it marks the node as 'failed' using a timestamp. When refreshing the server list via yb_servers(), if the driver sees a failed node in the response, it considers that server for new connections only if the time specified (in seconds) via this property has elapsed from the time it was last marked as failed. | 5 |
```
Comment (ddhodge) [inline docs/content/stable/drivers-orms/java/yugabyte-jdbc.md:129]:
```suggestion
| `failed-host-reconnect-delay-secs` | Time, in seconds, to wait before trying to connect to failed nodes. When the driver is unable to connect to a node, it marks the node as failed using a timestamp, and ignores the node when trying new connections until this time elapses. | 5 |
```

File: docs/content/preview/drivers-orms/java/yugabyte-jdbc.md [hunk 3]
```diff
@@ -125,18 +125,18 @@ The following table describes the connection parameters required to connect, inc
...
+| `user` | User connecting to the database | yugabyte |
+| `password` | User password | yugabyte |
+| `load-balance` | Enables [Uniform load balancing](../../smart-drivers/#cluster-aware-load-balancing) | false (Disabled) |
+| `yb-servers-refresh-interval` | The interval in seconds to refresh the servers list; ignored if `load-balance` is false | 300 |
+| `topology-keys` | Enables [Topology-aware load balancing](../../smart-drivers/#topology-aware-load-balancing). It can be set to comma-separated geo-locations in the form `cloud.region.zone:priority`. Ignored if `load-balance` is false | Empty |
+| `fallback-to-topology-keys-only` | If set to true and `topology-keys` are specified, the driver only tries to connect to nodes specified in `topology-keys` | false |
+| `failed-host-reconnect-delay-secs` | When the driver is unable to connect to a node, it marks the node as 'failed' using a timestamp. When refreshing the server list via yb_servers(), if the driver sees a failed node in the response, it considers that server for new connections only if the time specified (in seconds) via this property has elapsed from the time it was last marked as failed. | 5 |
```
Comment (ddhodge) [inline docs/content/preview/drivers-orms/java/yugabyte-jdbc.md:137]:
```suggestion
| `failed-host-reconnect-delay-secs` | Time, in seconds, to wait before trying to connect to failed nodes. When the driver is unable to connect to a node, it marks the node as failed using a timestamp, and ignores the node when trying new connections until this time elapses. | 5 |
```

File: docs/content/preview/drivers-orms/java/yugabyte-jdbc.md [hunk 4]
```diff
@@ -125,18 +125,18 @@ The following table describes the connection parameters required to connect, inc
...
+
+In v42.7.3-yb-1 and later, the `load-balance` property supports the following additional properties: any (alias for 'true'), only-primary, only-rr, prefer-primary, and prefer-rr. See [Node type aware load balancing](../../smart-drivers/#node-type-aware-load-balancing).
```
Comment (ddhodge) [inline docs/content/preview/drivers-orms/java/yugabyte-jdbc.md:139]:
```suggestion
In v42.7.3-yb-1 and later, the `load-balance` property supports the following additional properties: any (alias for 'true'), only-primary, only-rr, prefer-primary, and prefer-rr. See [Node type-aware load balancing](../../smart-drivers/#node-type-aware-load-balancing).
```

File: docs/content/preview/drivers-orms/smart-drivers.md [hunk 5]
```diff
@@ -188,25 +188,30 @@ If no servers are available, the request may return with a failure.
...
 The following shows how nodes are selected for new connections, depending on the load balance and topology key settings.
+The property names used are for the Java smart driver; the property names for other smart drivers may be slightly different.
+The property `fallback-to-topology-keys-only` is ignored when either the `topology-keys` empty or when `load-balance` is set to `prefer-primary` or `prefer-rr`.
 
 {{<tabpane text=true >}}
-{{% tab header="No topology keys" lang="no-keys" %}}
+{{% tab header="No topology-keys" lang="no-keys" %}}
 
-When no topology keys are specified, nodes are selected as follows.
+When no `topology-keys` is specified, nodes are selected as follows.
```
Comment (ddhodge) [inline docs/content/preview/drivers-orms/smart-drivers.md:197]:
```suggestion
When `topology-keys` is not specified, nodes are selected as follows.
```

File: docs/content/preview/drivers-orms/smart-drivers.md [hunk 6]
```diff
@@ -188,25 +188,30 @@ If no servers are available, the request may return with a failure.
 Consider a cluster with three primary nodes across three zones (zoneA, zoneB, and zoneC); and three read replica nodes in zoneB, zoneC, and zoneD.
 
 The following shows how nodes are selected for new connections, depending on the load balance and topology key settings.
+The property names used are for the Java smart driver; the property names for other smart drivers may be slightly different.
+The property `fallback-to-topology-keys-only` is ignored when either the `topology-keys` empty or when `load-balance` is set to `prefer-primary` or `prefer-rr`.
 
 {{<tabpane text=true >}}
```
Comment (ddhodge) [inline docs/content/preview/drivers-orms/smart-drivers.md:194]: The new tab setup doesn't seem like an improvement - there are 3 options, was the original found to be confusing?

Comment (ddhodge) [reply docs/content/preview/drivers-orms/smart-drivers.md]: @ashetkar

Comment (ashetkar) [reply docs/content/preview/drivers-orms/smart-drivers.md]: That's still incomplete @ddhodge, I have some changes not pushed yet. Please allow me another day to\ push those (seeing some formatting issues) and update this thread.

File: docs/content/v2.20/drivers-orms/java/yugabyte-jdbc.md (hunk 1 above)
Comment (ashetkar) [reply docs/content/v2.20/drivers-orms/java/yugabyte-jdbc.md:103]: This was inadvertent, removed

File: docs/content/preview/drivers-orms/smart-drivers.md (hunk 6 above)
Comment (ddhodge) [reply docs/content/preview/drivers-orms/smart-drivers.md]: @ashetkar Is it complete now?

Comment (ashetkar) [reply docs/content/preview/drivers-orms/smart-drivers.md]: Yes. I've pushed another commit today.

**Pull request merged**